databus-client2/src/main/java/com/bazaarvoice/emodb/databus/client2/discovery/EmoServiceDiscovery.java

```diff
@@ -0,0 +1,201 @@
+package com.bazaarvoice.emodb.databus.client2.discovery;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import com.google.common.io.Closeables;
+import com.google.common.util.concurrent.AbstractService;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.ChildData;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.utils.ZKPaths;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static com.bazaarvoice.emodb.databus.client2.Json.JsonUtil.parseJson;
+import static com.google.common.base.Preconditions.checkArgument;
+
+
+/**
+ * Service discovery for EmoDB.  This uses the same logic as in the EmoDB Java client but without the incompatible
+ * dependencies introduced by the current client.
+ */
+public abstract class EmoServiceDiscovery extends AbstractService {
```

Comment:
it would be great to split `EmoServiceDiscovery` into 2 classes, like
`DirectUrlEmoServiceDiscovery` and `ZookeeperEmoServiceDiscovery`